mne/time_frequency/spectrum.py

```diff
@@ -1226,11 +1197,21 @@ def __getitem__(self, item):
         return BaseRaw._getitem(self, item, return_times=False)
 
 
-def _check_data_shape(param, dim, data, expected):
-    if data.shape[dim] != expected:
+def _check_data_shape(data, freqs, info, ndim):
```

Comment:
I reworked this to check everything at once rather than needing multiple calls to the check function. Seemed simpler this way once we knew we only had to handle 2D and 3D cases.